src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -311,61 +296,40 @@ private void syncAcl() {
-                                                                     // worry
-                                                                     // about
-                                                                     // timeout?
-
-        if(logger.isDebugEnabled()) {
-            logger.debug("Read in rolesMapping {}", XContentHelper.convertToJson(response.getSourceAsBytesRef(), true, true));
-        }
-        return new SearchGuardRolesMapping().load(response.getSource());
-    }
-
-    private void writeACL(Client esClient, SearchGuardRoles roles, SearchGuardRolesMapping rolesMapping)
-            throws IOException {
-        IndexRequest rolesIR = new IndexRequest(searchGuardIndex).type(SEARCHGUARD_ROLE_TYPE).id(SEARCHGUARD_CONFIG_ID)
-                .refresh(true).consistencyLevel(WriteConsistencyLevel.DEFAULT).source(roles.toMap());
-        if(logger.isDebugEnabled()) {
-            logger.debug("Built roles request: {}", XContentHelper.convertToJson(rolesIR.source(),true, true));
-        }
-        String rID = esClient.index(rolesIR).actionGet().getId();
-        logger.debug("Roles ID: '{}'", rID);
-
-        IndexRequest mappingIR = new IndexRequest(searchGuardIndex).type(SEARCHGUARD_MAPPING_TYPE)
-                .id(SEARCHGUARD_CONFIG_ID).refresh(true).consistencyLevel(WriteConsistencyLevel.DEFAULT)
-                .source(rolesMapping.toMap());
-        if(logger.isDebugEnabled()) {
-            logger.debug("Built rolesMapping request: {}", XContentHelper.convertToJson(mappingIR.source(), true, true));
+    private void writeACL(PluginClient esClient, SearchGuardACLDocument... documents) throws Exception {
+        
+        BulkRequestBuilder builder = this.esClient.client().prepareBulk().setRefresh(true);
+        
+        for(SearchGuardACLDocument doc : documents) {
+            UpdateRequest update = esClient.client()
+                    .prepareUpdate(searchGuardIndex, doc.getType(), SEARCHGUARD_CONFIG_ID)
+                    .setConsistencyLevel(WriteConsistencyLevel.DEFAULT)
+                    .setDoc(doc.toXContentBuilder())
+                    .request();
+            builder.add(update);
+            if(logger.isDebugEnabled()) {
+                logger.debug("Built {} update request: {}", doc.getType(), XContentHelper.convertToJson(doc.toXContentBuilder().bytes(),true, true));
+            }
         }
-        String rmID = esClient.index(mappingIR).actionGet().getId();
-        logger.debug("rolesMapping ID: '{}'", rmID);
-
-        // force a config reload
-        ConfigUpdateResponse cur = esClient.execute(ConfigUpdateAction.INSTANCE, new ConfigUpdateRequest(SEARCHGUARD_INITIAL_CONFIGS))
-                .actionGet();
-
-        if (cur.getNodes().length > 0) {
-            logger.debug("Successfully reloaded config with '{}' nodes", cur.getNodes().length);
+        
+        BulkRequest request = builder.request();
```

Comment:
Whats the reasoning behind using a bulk request here? Because we aren't using the same kind of client as before?